### ISSUE TITLE
Add support for buildkit arguments

### ISSUE THREAD

**Issue by zoonage** [compose-spec/compose-spec#157]
**What is the problem you're trying to solve**
Add support for new buildkit arguments for docker-compose (e.g. SSH and secret mounts)

**Describe the solution you'd like**

**Additional context**s
https://github.com/compose-spec/compose-spec/pull/145

**Comment by ndeloof** [comment]
related: https://github.com/compose-spec/compose-spec/issues/81

**Comment by philomory** [comment]
So... this issue was opened to allow discussion, but so far (two months later), it... doesn't seem like there's been any actual discussion. Is anyone actually considering this?

**Comment by gc-ss** [comment]
>  Is anyone actually considering this?

I don't think so. After waiting, I am using this opportunity to build the image through `buildkit` using Hashicorp packer and then writing Hashicorp Nomad jobspecs instead of `docker-compose`.

Essentially migrating from `docker-compose` to Nomad jobspecs.

If you want to check out shipyard, that might be another option.

`docker-compose` has served me very well but I don't think there are plans to continue to move `docker-compose` forward with new features.

**Comment by ndeloof** [comment]
> I don't think there are plans to continue to move docker-compose forward with new features.

There is, but we are still focussing on getting Compose 2 out (compose in the Docker CLI, as a Go replacement for the legacy python docker-compose) and once this new codebase has reached parity it would be easier to introduce new features. See my talk by DockerCon "From Fig to Docker <space> Compose" #selfpromotion :P

**Comment by gc-ss** [comment]
> we are still focussing on getting Compose 2 out

Appreciate it. I am exclusively on the Compose 2 syntax. Never used the Compose 3 as I wasn't interested in Swarm.

Do you have an idea (nothing specific) when we can expect buildkit folded into Compose 2?

One of the reasons I started to use Compose was to do away with shell scripts but I'm having to turn to it so I can build using buildkit, so it looks like this:

```bash

#!/bin/bash -xv

# TODO: Use instead: http://dnephin.github.io/dobi/
export DOCKER_BUILDKIT=1
export COMPOSE_DOCKER_CLI_BUILD=1
# NOTE: If you use a different ssh key to auth with the code repo, replace id_rsa with that
docker build --file Dockerfile.ext --ssh bitbucket=$HOME/.ssh/id_rsa --tag blah/blah:2021-04-29 .  # Now run the build
# give it a shot and see if the image built properly!
docker-compose run --service-ports --rm sanity_check; docker-compose down
```

I don't think that `export COMPOSE_DOCKER_CLI_BUILD=1` is meaningful right now.

Not complicated, but not as nice as just a compose file!

**Comment by EricHripko** [comment]
There's a bit of miscommunication at play - I believe @ndeloof meant new `docker compose` Go CLI that his team is working on. There's a discussion on the build-time secrets already under #81 - feel free to contribute there.

**Comment by ndeloof** [comment]
yes indeed, sorry for the confusion. We definitively should have found a better name for "Compose v2". "Compose.Next"? "Compose.NewGen"? Naming things without ambiguity is one of the harder problem in our industry :P
